src/Workspaces/Core/Portable/Workspace/Solution/Checksum.cs

```diff
@@ -20,16 +20,18 @@ namespace Microsoft.CodeAnalysis
     /// Checksum of data can be used later to see whether two data are same or not
     /// without actually comparing data itself
     /// </summary>
-    [DataContract]
-    internal sealed partial record class Checksum(
-        [property: DataMember(Order = 0)] Checksum.HashData Hash)
+    [DataContract, StructLayout(LayoutKind.Explicit, Size = HashSize)]
+    internal readonly partial record struct Checksum(
+        [field: FieldOffset(0)][property: DataMember(Order = 0)] long Data1,
+        [field: FieldOffset(8)][property: DataMember(Order = 1)] long Data2,
+        [field: FieldOffset(16)][property: DataMember(Order = 2)] int Data3)
     {
         /// <summary>
-        /// The intended size of the <see cref="HashData"/> structure. 
+        /// The intended size of the <see cref="Checksum"/> structure. 
         /// </summary>
         public const int HashSize = 20;
 
-        public static readonly Checksum Null = new(Hash: default);
+        public static readonly Checksum Null = default;
```

Comment:
>Null [](http://example.com/codeflow?start=40&length=4)

`Default`? Seems odd to call it `Null` when the type is a struct. Or perhaps remove entirely and replace with `default`.